rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -19,164 +19,332 @@
-    private FederationConstants federationMainnetConstants;
-    private NetworkParameters networkParameters;
+    private static final FederationConstants federationMainnetConstants = FederationMainNetConstants.getInstance();
+    private final Federation genesisFederation = FederationTestUtils.getGenesisFederation(federationMainnetConstants);
+    private Federation oldFederation;
+    private Federation newFederation;
     private FederationStorageProvider storageProvider;
-    private FederationSupportBuilder federationSupportBuilder;
+    private FederationSupportBuilder federationSupportBuilder = new FederationSupportBuilder();
+    private FederationSupport federationSupport;
 
     @BeforeEach
     void setUp() {
         StorageAccessor storageAccessor = new InMemoryStorage();
         storageProvider = new FederationStorageProviderImpl(storageAccessor);
-        federationMainnetConstants = FederationMainNetConstants.getInstance();
-        networkParameters = federationMainnetConstants.getBtcParams();
-        federationSupportBuilder = new FederationSupportBuilder();
 
         federationSupport = federationSupportBuilder
             .withFederationConstants(federationMainnetConstants)
             .withFederationStorageProvider(storageProvider)
             .build();
     }
 
-    @Test
-    void getActiveFederation_withNullFederations_returnsGenesisFederation() {
-        Federation genesisFederation = FederationTestUtils.getGenesisFederation(federationMainnetConstants);
-        Federation activeFederation = federationSupport.getActiveFederation();
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    @Tag("getActiveFederationTestsWithNullFederations")
+    class ActiveFederationTestsWithNullFederations {
+        @Tag("getActiveFederationTestsWithNullFederations")
+
+        @BeforeAll
+        void setUp() {
+            StorageAccessor storageAccessor = new InMemoryStorage();
+            storageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+            federationSupportBuilder = new FederationSupportBuilder();
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .build();
+        }
 
-        assertThat(activeFederation, is(genesisFederation));
-    }
+        @Test
+        void getActiveFederation_returnsGenesisFederation() {
+            Federation activeFederation = federationSupport.getActiveFederation();
+            assertThat(activeFederation, is(genesisFederation));
+        }
 
-    @Test
-    void getActiveFederation_withOldFederationAndNullNewFederation_returnsGenesisFederation() {
-        Federation oldFederation = FederationTestUtils.getErpFederation(networkParameters);
-        storageProvider.setOldFederation(oldFederation);
+        @Test
+        void getActiveFederationRedeemScript_beforeRSKIP293_returnsEmpty() {
```

Comment:
this was added too to prove this happens also in this context (not only when having two null feds)